keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -277,18 +276,18 @@ class ApplicationService(
     }
 
     val artifactSummaries = deliveryConfig.artifacts.map { artifact ->
+      // We calculate a bunch of data for each artifact, so pre-load the bulk of it here and pass that info into
+      //  later functions to save redundant database calls.
       val artifactVersions = repository.artifactVersions(artifact, limit)
+      val artifactVersionInfoInEnvironment: Map<String, List<StatusInfoForArtifactInEnvironment>> = deliveryConfig.environments.map { env ->
+        env.name to repository.getVersionInfoInEnvironment(deliveryConfig, env.name, artifact)
+      }.toMap()
 
-      val verificationStartTime = now
       // A verification context identifies an artifact version in an environment
       // For each context, there may be multiple verifications (e.g., test-container, canary)
       //
       // This map associates a context with this collection of verifications and their states
       val verificationStateMap = getVerificationStates(deliveryConfig, artifactVersions)
-      spectator.timer(
-        VERIFICATION_SUMMARY_CONSTRUCT_DURATION_ID,
-        listOf(BasicTag("application", application))
-      ).record(Duration.between(verificationStartTime, now))
```

Comment:
This didn't seem worthwhile to measure (so little time) so I'm removing the metric.